src/Aspire.Cli/Mcp/WaitForResourceStateTool.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using ModelContextProtocol;
+using ModelContextProtocol.Protocol;
+
+namespace Aspire.Cli.Mcp;
+
+internal sealed class WaitForResourceStateTool : CliMcpTool
+{
+    public override string Name => "wait_for_resource_state";
+
+    public override string Description => "Waits for a resource to enter a specific state. The tool will wait up to 30 seconds for the resource to reach the desired state. Valid states include: Running, Starting, Exited, FailedToStart, Finished, Building, Waiting, Stopping, Unknown, RuntimeUnhealthy, NotStarted, Hidden.";
```

Comment:
Would it be useful to provide a description of what the states mean? Or to simplify this list?

Most of them are pretty self explanitory, but the difference between Exited and Finished isn't obvious.